### PR TITLE
feat(jdk21): add the tooling installation for jdk21 beta

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3709

### CODE REVIEW

**Review by timja — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Note: for this role, there may be the following error message because plugins in
 
 You can safely ignore it.
 
-You can re-run puppet and execute tests with `vagrant provision ROLE` repeatedly while the VM is up and running.
+You can re-run puppet and execute tests with `vagrant provision ROLE` repeatedly while the VM is up and running or `vagrant up jenkins::controller --provision
```
Comment (lemeurherve) [inline README.md]:
```suggestion
You can re-run puppet and execute tests with `vagrant provision ROLE` repeatedly while the VM is up and running, or `vagrant up jenkins::controller --provision`
```

File: dist/profile/templates/jenkinscontroller/casc/tools.yaml.erb [hunk 2]
```diff
@@ -56,8 +56,8 @@ tool:
             }
             -%>
               # subdir is the top-level directory in the archive: depends on the how the JDK is packaged. Defaults to adoptium convention unless a customized version is specified
-              # Example of subdirs for adoptium 8: jdk8u345-b01, adoptium 11: jdk-11.0.16+8, adoptium 17: jdk-17.0.4+8 and adoptium 19: jdk-19.0.1+10
-              subdir: "<%= installer_setup['subdir'] ? installer_setup['subdir'] : ("jdk" + ($jdk_major_version.to_i == 8 ? '' : '-') + $jdk['version']) %>"
+              # Example of subdirs for adoptium 8: jdk8u345-b01, adoptium 11: jdk-11.0.16+8, adoptium 17: jdk-17.0.4+8, adoptium 19: jdk-19.0.1+10, adoptium 21: jdk-2023-08-09-06-56-beta ...
```
Comment (lemeurherve) [inline dist/profile/templates/jenkinscontroller/casc/tools.yaml.erb]:
```suggestion
              # Example of subdirs for adoptium 8: jdk8u345-b01, adoptium 11: jdk-11.0.16+8, adoptium 17: jdk-17.0.4+8, adoptium 19: jdk-19.0.1+10, adoptium 21: jdk-2023-08-09-06-56-beta
```

**Review by lemeurherve — DISMISSED** [review]
Some nits, LGTM!

File: README.md (hunk 1 above)
Comment (smerle33) [reply README.md]: this change was useless so I removed it

File: dist/profile/templates/jenkinscontroller/casc/tools.yaml.erb (hunk 2 above)
Comment (smerle33) [reply dist/profile/templates/jenkinscontroller/casc/tools.yaml.erb]: this path was wrong

**Comment by smerle33** [comment]
after a pair programming with @dduportal I moved the version definition for jdk21 to the installer part, as long as it's a beta. The name differ without a `-` and a `beta` in the path ... also the zip file main folder is jdk-21+34 and will probably change over time so better to keep it in the installer tools area vs the `tools_default_versions`

File: hieradata/clients/controller.ci.jenkins.io.yaml [hunk 3]
```diff
@@ -116,6 +116,18 @@ profile::jenkinscontroller::jcasc:
             type: "zip"
             label: "s390x"
             cpu_arch: "s390x"
+      jdk21:
+        installers:
+          linux-arm64:
+            type: "zip"
+            label: "linux && arm64"
```
Comment (dduportal) [inline hieradata/clients/controller.ci.jenkins.io.yaml:123]:
```suggestion
            label: "linux && arm64"
            subdir: "jdk-21+34"
```

File: hieradata/clients/controller.ci.jenkins.io.yaml [hunk 4]
```diff
@@ -116,6 +116,18 @@ profile::jenkinscontroller::jcasc:
...
+            cpu_arch: "aarch64"
+            version: "21-2023-08-09-06-56-beta"
+          s390x:
+            type: "zip"
+            label: "s390x"
```
Comment (dduportal) [inline hieradata/clients/controller.ci.jenkins.io.yaml:129]:
```suggestion
            label: "s390x"
            subdir: "jdk-21+34"
```

File: hieradata/vagrant/common.yaml [hunk 5]
```diff
@@ -94,6 +94,15 @@ profile::jenkinscontroller::jcasc:
             type: "zip"
             label: "s390x"
             cpu_arch: "s390x"
+      jdk21:
+        installers:
+          linux-amd64:
+            type: "zip"
+            label: "linux && amd64"
```
Comment (dduportal) [inline hieradata/vagrant/common.yaml]:
```suggestion
          linux-arm64:
            type: "zip"
            label: "linux && arm64"
            subdir: "jdk-21+34"
            version: "21-2023-08-09-06-56-beta"
```

File: hieradata/vagrant/common.yaml [hunk 6]
```diff
@@ -94,6 +94,15 @@ profile::jenkinscontroller::jcasc:
...
+          windows-amd64:
+            type: "zip"
+            label: "windows"
+            os: "windows"
```
Comment (dduportal) [inline hieradata/vagrant/common.yaml]:
```suggestion
```

**Review by dduportal — CHANGES_REQUESTED** [review]
2 blockers:

- Don't forget the subdir on all installation kinds
- Vagrant's hieradata is like ci.j's hieradata: it inherits from the `hieradata/common`: no need to re-define the amd64 linux and windows. But I've added linux arm64 to allow testing on apple silicon (+ inheriting from the amd64 from parent common)

**Review by dduportal — APPROVED** [review]

**Pull request merged**